### ISSUE TITLE
Purpose of repository contents

### ISSUE THREAD

**Issue by bratekarate** [FIGLAB/hand-activities#3]
Thanks a lot for sharing this repository as an addition to your research paper. 

I am, however, not sure if I understand the purpose of the software in this repository. My presumptions and questions so far:
- The `kernmod` directory contains assets to overclock the smartwatch processor. Contents:
    - script that downloads kernel images (what is the difference between those images? Should the `boot-new.img` be used with fastboot to prepare the device?)
    - fastboot binaries (is there a difference to other fastboot binaries i could install from elsewhere?)
    - `fastacc_mpu.ko`, kernel image to load in the smartwatches' kernel (after `.img` has been loaded with fastboot?)
- The `backend` directory contains a python server for the backend server. It is accessed via TCP/HTTP. But why does it say in `server.py` "Python server to receive data from one or more HandActivities Server"? Isn't this very program the HandActivitiesServer? I would presume it receives data from HandActivities _Clients_. Please correct me if I'm wrong.
- The file `model_main.hdf5` required by `model.py` contains the data model for the classification of gestures. Will this data be shared in the future as well?

Guidance would be greatly appreciated as I plan to build upon this research with my master's thesis.

**Comment by AutonomicPerfectionist** [comment]
I've been following this project and related ones for awhile now, from what I can tell this project is a continuation of Viband and appears to use a similar kernel driver found in FastAcc. There is no public client code that I have found, although I have successfully written a Python test program to extract the necessary data, called Viband-python, using the information found in their published paper. The program is incomplete as I no longer have the time or resources to dedicate to it, but you are welcome to take a look at it. Using it requires a Debian chroot environment on the watch as it is a simple Python program not adhering to Android's app framework.

As for the model, they have said it will be shared but it has been a significantly long time since then, with no word on it and no subsequent commits. Their papers do have information that could be used to train a model yourself however.

**Comment by bratekarate** [comment]
> I've been following this project and related ones for awhile now, from what I can tell this project is a continuation of Viband and appears to use a similar kernel driver found in FastAcc

Yes, I have read parts of the corresponding paper briefly. This repository is linked in a paper that builds upon the original ViBand project. I have also found the FastAcc repository and was wondering if the kernel driver is actually the same.

> There is no public client code that I have found, although I have successfully written a Python test program to extract the necessary data, called Viband-python, using the information found in their published paper

So it is as I expected only the server code. But your project sounds very promising, I will definitely check it out. Thanks a lot for letting me know!

> As for the model, they have said it will be shared but it has been a significantly long time since then, with no word on it and no subsequent commits

I read about this in the readme but was not sure if I understood correctly that they plan to release the model file for the python program. Since some time has passed already, it is probably better not to count on it.

The model, however, is not that important for me anyway. In my thesis I want to collect gesture data from a smartwatch, send it via bluetooth to a smartphone, just as they described in their paper, but then send it to a Learning Analytics engine that has been developed in an earlier project. The analytics and machine learning part is not the main focus of my project. I am rather concerned with preprocessing the data and feeding it into a Kafka topic.

Anyway, thank you again for your help!
